lib/matplotlib/axes/_axes.py

```diff
@@ -6142,6 +6147,16 @@ def _normalize_input(inp, ename='input'):
                 raise ValueError(
                     'weights should have the same shape as x')
 
+        # Combine the masks from x[i] and w[i] (if applicable) into a single
+        # mask and apply it to both.
+        if not input_empty:
+            for i in range(len(x)):
+                mask_i = x[i].mask
+                if w[i] is not None:
+                    mask_i = mask_i | w[i].mask
```

Comment:
I would raise an error if the weight matrix contained mask elements that weren't masked in the data `x` itself. Unless I am not seeing a use case for that?